arangod/RocksDBEngine/RocksDBZkdIndex.cpp

```diff
@@ -579,10 +578,36 @@ auto zkd::supportsFilterCondition(
     return {};  // all prefix values have to be assigned
   }
 
-  // TODO -- actually return costs
-  auto costs = Index::FilterCosts::defaultCosts(itemsInIndex);
-  costs.coveredAttributes = extractedBounds.size() + extractedPrefix.size();
+  Index::FilterCosts costs;
   costs.supportsCondition = true;
+  costs.coveredAttributes = extractedBounds.size() + extractedPrefix.size();
+
+  // we look up a single point using the prefix values
+  const double estimatedElementsOnCurve =
+      index->hasSelectivityEstimate() ? 1. / index->selectivityEstimate()
+                                      : static_cast<double>(itemsInIndex);
+
+  // each additional bound reduces the volume
+  const double volumeReductionFactor = 1.4;  // guessed, 2 might be too much
+  const double searchBoxVolume =
+      1. /
+      pow(volumeReductionFactor, static_cast<double>(extractedBounds.size()));
+
+  costs.estimatedItems =
+      static_cast<size_t>(estimatedElementsOnCurve * searchBoxVolume);
+
+  const size_t unusedDimensions =
+      index->fields().size() - extractedBounds.size();
```

Comment:
I guess `extractedBounds` is always `<=` to `index->fields().size()` here, but I am not 100% sure.
It is fine if this is already guarenteed by other means.

Reply:
yes, it is guaranteed.